src/charm.py

```diff
@@ -948,13 +1080,59 @@ def _generate_vault_config_file(self) -> None:
             node_id=self._node_id,
             retry_joins=retry_joins,
         )
+        content = self._add_transit_stanza_if_needed(content)
         existing_content = ""
         if self._container.exists(path=VAULT_CONFIG_FILE_PATH):
             existing_content_stringio = self._container.pull(path=VAULT_CONFIG_FILE_PATH)
             existing_content = existing_content_stringio.read()
 
         if not config_file_content_matches(existing_content=existing_content, new_content=content):
             self._push_config_file_to_workload(content=content)
+            if _contains_transit_stanza(existing_content) != _contains_transit_stanza(content):
+                if self._vault_service_is_running():
+                    self._container.restart(self._service_name)
```

Comment:
Why is the restart needed?

Reply:
I mean why this does, but a change in the config file not related to the transit stanza doesn't?

Reply:
That's a good question and something I should document.

Changing the seal type requires restarting the node to trigger the changes. It just so happens that transit is the only other seal type we have. This is changing, v1.16 has a beta feature which won't require this.

Reply:
Oops, I didn't notice that this beta feature is only for Enterprise. :cry:

Reply:
@saltiyazan Can you let me know if this helps? https://github.com/canonical/vault-k8s-operator/pull/365/commits/87c4ba30a147cabc9bebd7bfafb820386c444c48